core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -100,19 +100,20 @@ object ByteBufferMessageSet {
       }
       var lastInnerOffset = -1L
 
-      val messageAndOffsets = if (wrapperMessageAndOffset.message.magic > MagicValue_V0) {
+      val messageAndOffsets = {
         val innerMessageAndOffsets = new ArrayDeque[MessageAndOffset]()
         try {
           while (true)
             innerMessageAndOffsets.add(readMessageFromStream())
         } catch {
           case eofe: EOFException =>
-            compressed.close()
           case ioe: IOException =>
             throw new InvalidMessageException(s"Error while reading message from stream compressed with ${wrapperMessage.compressionCodec}", ioe)
+        } finally {
+          compressed.close()
```

Comment:
Maybe we want to do `CoreUtils.swallow(compressed.close())` to avoid a potential exception in `close` obscuring an exception in the `try`.